plan_delegator/src/plan_delegator.cpp

```diff
@@ -422,78 +453,82 @@ namespace plan_delegator
             auto maneuver_planner = GET_MANEUVER_PROPERTY(maneuver, parameters.planning_tactical_plugin);
             auto client = getPlannerClientByName(maneuver_planner);
             
-            ROS_DEBUG_STREAM("Current planner: " << maneuver_planner);
+            RCLCPP_DEBUG_STREAM(rclcpp::get_logger("plan_delegator"),"Current planner: " << maneuver_planner);
 
             // compose service request
             auto plan_req = composePlanTrajectoryRequest(latest_trajectory_plan, current_maneuver_index);
-
-            if(client.call(plan_req))
+            
+            auto plan_response = client->async_send_request(plan_req);
+            
+            // Wait for the result.
+            if (rclcpp::spin_until_future_complete(shared_from_this(), plan_response) ==
```

Comment:
spin_until_future_complete will allow for topic callbacks to occur while waiting for the response which may cause strange  multi-threading issues.  I would suggest using future.wait_for() instead see https://github.com/usdot-fhwa-stol/carma-utils/blob/4adc6cd3c8169879e037359b64d7ee809d0f3c42/ros2_lifecycle_manager/src/ros2_lifecycle_manager.cpp#L209

CarmaLifecycleNode has special automatic settings for service calls to allow for them to return without having to spin the topic queue to allow for a single threaded approach like in ros1.